src/Microsoft.ML.Data/Scorers/BinaryClassifierScorer.cs

```diff
@@ -63,7 +63,7 @@ private static ISchemaBoundMapper WrapIfNeeded(IHostEnvironment env, ISchemaBoun
 
             if (trainSchema?.Label == null)
                 return mapper; // We don't even have a label identified in a training schema.
-            var keyType = trainSchema.Label.Value.Metadata.Schema.GetColumnOrNull(MetadataUtils.Kinds.KeyValues)?.Type;
+            var keyType = trainSchema.Label.Value.Metadata.Schema.GetColumnOrNull(MetadataUtils.Kinds.KeyValues)?.Type as VectorType;
```

Comment:
>VectorType [](start = 122, length = 10)

why?

Reply:
`CanWrap` checks whether this column is a `VectorType`, so this is "safe". Doing the cast was necessary for line 71's usage of `keyType.ItemType`.

---
In reply to: [247769095](https://github.com/dotnet/machinelearning/pull/2131#discussion_r247769095) [](ancestors = 247769095)